core/src/main/java/hudson/model/Job.java

```diff
@@ -238,6 +239,30 @@ public void onLoad(ItemGroup<? extends Item> parent, String name)
 
         for (JobProperty p : properties)
             p.setOwner(this);
+
+        if (this instanceof LazyBuildMixIn.LazyLoadingJob) {
+            // LazyBuildMixIn.onLoad will call fixNextBuildNumber() at just the right time
+        } else {
```

Comment:
The only case I know of where this clause would be triggered is if you are using the `external-monitor-job` plugin, which has its own weird numbering system anyway.